ios/brave-ios/Sources/Brave/Frontend/Browser/New Tab Page/NewTabPageViewController.swift

```diff
@@ -435,6 +455,120 @@ class NewTabPageViewController: UIViewController {
+        self?.collectionView.alpha = 1
+      }
+    )
+  }
+
+  private func showAndFadeInCollectionView() {
+    collectionView.isHidden = false
+    collectionView.alpha = 0
+    UIView.animate(
+      withDuration: 0.3,
+      animations: { [weak self] in
+        self?.collectionView.alpha = 1
+      }
+    )
+  }
+
+  func setupBackgroundVideo() {
+    videoButtonsView.isHidden = true
+
+    guard let backgroundVideoPath = background.backgroundVideoPath else {
+      return
+    }
+    videoPlayer = NewTabPageVideoPlayer()
+
+    videoButtonsView.tappedBackgroundVideo = { [weak videoPlayer] in
+      guard let videoPlayer else {
+        return false
+      }
+      return videoPlayer.togglePlay()
+    }
+    videoButtonsView.tappedCancelButton = { [weak videoPlayer] in
+      videoPlayer?.maybeCancelPlay()
+    }
+
+    backgroundButtonsView.activeButton = .none
+    backgroundButtonsView.tappedPlayButton = { [weak self] in
+      guard let self = self else { return }
+      self.videoButtonsView.isHidden = false
+      self.fadeOutAndHideCollectionView()
+      self.videoPlayer?.startPlayback()
+    }
+    backgroundButtonsView.tappedBackgroundDuringAutoplay = { [weak self] in
+      guard let self = self else { return }
+      self.collectionView.isHidden = true
+      self.videoButtonsView.isHidden = false
+      self.videoPlayer?.startPlayback()
+    }
+
+    videoPlayer?.didCancelPlaybackEvent = { [weak self] in
+      self?.videoButtonsView.isHidden = true
+      self?.showAndFadeInCollectionView()
+    }
+    videoPlayer?.didStartAutoplayEvent = { [weak self] in
+      self?.backgroundButtonsView.videoAutoplayStarted()
+    }
+    videoPlayer?.didFinishAutoplayEvent = { [weak self] in
+      guard let self = self else { return }
+      self.backgroundButtonsView.videoAutoplayFinished()
+      if case .sponsoredImage(let background) = self.background.currentBackground {
+        self.backgroundButtonsView.activeButton = .brandLogo(background.logo)
+      }
+      self.backgroundButtonsView.alpha = 0
+      UIView.animate(
+        withDuration: 0.3,
+        animations: { [weak self] in
+          self?.backgroundButtonsView.alpha = 1
+        }
+      )
+    }
+    videoPlayer?.didFinishPlaybackEvent = { [weak self] in
+      guard let self = self else { return }
+      self.videoButtonsView.isHidden = true
+      self.reportSponsoredBackgroundEvent(.media100)
+      self.showAndFadeInCollectionView()
+    }
+    videoPlayer?.didStartPlaybackEvent = { [weak self] in
+      self?.reportSponsoredBackgroundEvent(.mediaPlay)
+    }
+    videoPlayer?.didPlay25PercentEvent = { [weak self] in
+      self?.reportSponsoredBackgroundEvent(.media25)
+    }
+
+    videoPlayer?.setupPlayer(backgroundVideoPath: backgroundVideoPath)
+    backgroundView.playerLayer.player = videoPlayer?.player
+    backgroundView.setupPlayerLayer(backgroundVideoPath)
+  }
+
+  private func shouldShowVideoBackground() -> Bool {
```

Comment:
nit: `shouldShowBackgroundVideo` to match other usages of `...BackgroundVideo...`